### PR TITLE
fix(topology): fix code smells

### PR DESCRIPTION
**Pull request by debsmita1**
**Resolves:**
https://github.com/janus-idp/backstage-plugins/issues/587

### CODE REVIEW

File: plugins/topology/src/utils/workload-node-utils.ts [hunk 1]
```diff
@@ -132,7 +132,7 @@ export const calculateRadius = (size: number) => {
 
 const getScalingUp = (dc: K8sWorkloadResource) => {
   return {
-    ...(dc.metadata && dc.metadata),
+    ...(dc.metadata ? dc.metadata : {}),
```
Comment (tumido) [inline plugins/topology/src/utils/workload-node-utils.ts:135]:
```suggestion
    ...(dc.metadata || {}),
```

File: plugins/topology/src/utils/pod-resource-utils.ts [hunk 2]
```diff
@@ -354,7 +348,7 @@ export const getPodsForDaemonSet = (
 export const getPodsDataForResource = (
   resource: K8sWorkloadResource,
   resources: K8sResponseData,
-): PodRCData => {
```
Comment (divyanshiGupta) [inline plugins/topology/src/utils/pod-resource-utils.ts:357]: @debsmita1  Why was this return type removed?

**Review by divyanshiGupta — CHANGES_REQUESTED** [review]

File: plugins/topology/src/utils/resource-utils.ts [hunk 3]
```diff
@@ -60,18 +60,18 @@ export const createOverviewItemForType = (
...
         ? {
-            obj: resource as K8sWorkloadResource,
+            obj: resource,
           }
         : undefined;
     case 'pods':
       return validPod(resource as V1Pod)
         ? {
-            obj: resource as K8sWorkloadResource,
+            obj: resource,
```
Comment (divyanshiGupta) [inline plugins/topology/src/utils/resource-utils.ts:69]: @debsmita1 why is the type removed here?

Comment (debsmita1) [reply plugins/topology/src/utils/resource-utils.ts:69]: Because the type of `resource` parameter is already defined above

**Review by divyanshiGupta — APPROVED** [review]
LGTM!

**Review by tumido — CHANGES_REQUESTED** [review]

**Review by tumido — APPROVED** [review]
/lgtm
/approve

**Pull request merged**